routes/spots.js

```diff
@@ -1,19 +1,57 @@
-const express = require('express');
-const { Spot, Review } = require('../db/models');
+const express = require("express");
+const { Spot, Review, Collection } = require("../db/models"); // Changed
+const { Op } = require("sequelize"); //Changed
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
-    res.redirect('/');
+router.get("/", (req, res) => {
+  res.redirect("/");
 });
 
-router.get('/:id', async (req, res, next) => {
-    const id = req.params.id;
+router.get("/:id", async (req, res, next) => {
+  const id = req.params.id;
+  const userId = req.session.auth.userId;
 
-    const spot = await Spot.findByPk(id);
-    const reviews = await Review.findAll({where: { spotId: id}});
+  const spot = await Spot.findByPk(id);
+  // const reviews = await Review.findAll({ where: { spotId: id } });
 
-    res.render('spot', { title: `${spot.name}`, spot, reviews, user: req.session.user });
-})
+  // Optimize the above query to do eager load of Review
+
+
+  const customCollections = await Collection.findAll({
+    where: { userId: userId },

```

Comment:
Heads up: I ran this code without being an authenticated user and it threw an error. Make sure to you build a check into your logic to determine if the user is signed in. If they aren't then don't send the collections in the response. To replicate the error try going to a spot landing page such as `http://localhost:8080/spots/6` while browsing the site unauthenticated. I.e., `req.session.auth.userId` is undefined if the user is not logged in.